### PR TITLE
Add aggregated search paths targets to vendored build settings

### LINKED ISSUE #5512: 1.0.1: Missing framework search path on Unit Test target
- [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).
# Report
## What did you do?

Sample project available here: https://github.com/plu/SearchPath

`Podfile`

``` ruby
source 'https://github.com/CocoaPods/Specs'

use_frameworks!

target 'SearchPath' do
  platform :ios, '8.0'

  pod '1PasswordExtension'
  pod 'Google-Mobile-Ads-SDK'

  target 'SearchPathTests' do
    inherit! :search_paths
  end
end
```

`SearchPathTests.swift`

``` swift
import XCTest
import OnePasswordExtension // OK
import GoogleMobileAds // NOK: No such module GoogleMobileAds
@testable import SearchPath

class SearchPathTests: XCTestCase {

    func testExample() {
        XCTAssert(true)
    }

}
```
## What did you expect to happen?

That it can compile and execute `SearchPathTests.swift`.
## What happened instead?

It fails to compile, error `No such module GoogleMobileAds`. However it can find the `OnePasswordExtension` module. I'd expect the framework search path to be the same for the app target and the test target, when I specify `inherit! :search_paths` in the `Podfile`. I'm not sure why it would make a difference between `OnePasswordExtension` and `GoogleMobileAds`. In the attached screenshots you can see that the `SearchPathTests` target is missing `"$(PODS_ROOT)/Google-Mobile-Ads-SDK/Frameworks"` but contains `"$PODS_CONFIGURATION_BUILD_DIR/1PasswordExtension"`.
### Stack

```
   CocoaPods : 1.0.1
        Ruby : ruby 2.1.2p95 (2014-05-08 revision 45877) [x86_64-darwin13.0]
    RubyGems : 2.5.2
        Host : Mac OS X 10.11.5 (15F34)
       Xcode : 7.3.1 (7D1014)
         Git : git version 2.7.4 (Apple Git-66)
Ruby lib dir : /Users/joplunien/.rvm/rubies/ruby-2.1.2/lib
Repositories : ebay-cocoapods-specs - git@github.corp.ebay.com:eBay-Kleinanzeigen/cocoapods-specs.git @ cf942c64824c2afefcd668b8d4203a341799dcda
               master - https://github.com/CocoaPods/Specs @ d7e019e4b893db3e9a8b198599d5909e80fdf127
```
### Installation Source

```
Executable Path: /Users/plu/.rvm/gems/ruby-2.1.2/bin/pod
```
### Plugins

```
cocoapods-deintegrate      : 1.0.0
cocoapods-plugins          : 1.0.0
cocoapods-search           : 1.0.0
cocoapods-stats            : 1.0.0
cocoapods-superdeintegrate : 0.0.1
cocoapods-trunk            : 1.0.0
cocoapods-try              : 1.0.0
```
### Podfile

``` ruby
source 'https://github.com/CocoaPods/Specs'

use_frameworks!

target 'SearchPath' do
  platform :ios, '8.0'

  pod '1PasswordExtension'
  pod 'Google-Mobile-Ads-SDK'

  target 'SearchPathTests' do
    inherit! :search_paths
  end
end
```
## Project that demonstrates the issue

https://github.com/plu/SearchPath

app_target
test_target
test_target_build_failure

### PR DESCRIPTION
**Pull request by plu**
Fix #5512 

I assume this will break some tests. Before fixing/adapting them I'd like to see your approval that this is the correct fix :).

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
        
            1 Error
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><p>Please include a CHANGELOG entry to credit yourself! 
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</p>
</td>
    </tr>
    
    
  </tbody>
</table>
  

  

  



Here's an example of your CHANGELOG entry:

```markdown
* Add aggregated search paths targets to vendored build settings 
  [plu](https://github.com/plu)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb [hunk 1]
```diff
@@ -150,7 +150,9 @@ def generate_settings_to_import_pod_targets
         #   user target.
         #
         def generate_vendored_build_settings
-          pod_targets.each do |pod_target|
+          targets = pod_targets + target.search_paths_aggregate_targets.map(&:pod_targets)
```
Comment (segiddins) [inline lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb]: This should use `flat_map`

**Comment by segiddins** [comment]
@mrackwitz looks good to me, seem good to you too?

**Comment by plu** [comment]
I'll have a look at the failing tests in the meantime.

**Comment by plu** [comment]
I'm using `Rainforest` locally, but I have a list of failures on `master`:

```diff
-    - '[CP] Check Pods Manifest.lock': []
+    - "[CP] Check Pods Manifest.lock": []
```

Do you have any hint for me where this might be coming from?

**Comment by DanToml** [comment]
@plu Those should pass if you rebase on master, tests on Ruby >= 2.3 were fixed in https://github.com/CocoaPods/CocoaPods/pull/4687

**Comment by plu** [comment]
I was using `2.1.2p95` before, now I'm trying with `2.3.1p112`, both unsuccessful. I'm on this commit in `spec/cocoapods-integration-specs`:

```
commit 8d95e03ca3c6e3e2e8f0d870720110569ed3f45a (HEAD -> master, origin/master, origin/HEAD)
Merge: 4c7f241 409958a
Author: Samuel Giddins <segiddins@segiddins.me>
Date:   Fri Jun 10 00:35:41 2016 -0500

    Merge pull request #72 from Monits/circular_dependency

    [PodTarget] Add a cyclic dependencies between subspecs case
```

**Comment by plu** [comment]
Ok, rebasing `CocoaPods.git` upon `master` worked - thanks :)

**Comment by plu** [comment]
I've updated the test expectations. I'm not sure how to handle the submodule reference - am I supposed to update that one?

**Comment by mrackwitz** [comment]
It would be nice to have a unit test for that. Beside that this looks good to me though! :+1:
You don't have to worry about updating the submodule.

**Comment by plu** [comment]
I assume the test should go into `spec/unit/generator/xcconfig/aggregate_xcconfig_spec.rb`, but to be honest, I'm not sure how to create the nested targets with `inherit! :search_path` scenario there.

**Comment by benasher44** [comment]
This might also fix #5173

**Comment by Coeur** [comment]
@plu, this branch has conflicts
@mrackwitz, if it's too complex to write a test case, may we still merge the pull request for 1.1.0.beta.2 phase?

**Comment by mrackwitz** [comment]
>if it's too complex to write a test case, may we still merge the pull request for 1.1.0.beta.2 phase?

@Coeur: I'm asking mainly for tests not because I'm afraid that these changes here could be wrong or have unexpected effects, but more because untested code tends to break in the future when you're unaware because expectations were not encoded in the specs. So this is completely unrelated from the release context. If that makes it in without tests, no one is ever going to write tests for it.

**Comment by esetnik** [comment]
I think I just got bit by #5512, hoping to see this make it into 1.1.0. Let me know if I can help test.

**Comment by chrisortman** [comment]
Testing this branch and I get an error trying to install the KeychainAccess pod

```
NoMethodError - undefined method `try' for <Pod::PodTarget name=KeychainAccess >:Pod::PodTarget
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/generator/xcconfig/xcconfig_helper.rb:287:in `add_language_specific_settings'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/generator/xcconfig/xcconfig_helper.rb:228:in `add_target_specific_settings'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/generator/xcconfig/pod_xcconfig.rb:65:in `generate'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/generator/xcconfig/pod_xcconfig.rb:35:in `save_as'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:207:in `create_xcconfig_file'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:24:in `block in install!'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/user_interface.rb:141:in `message'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:19:in `install!'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer/xcode/pods_project_generator.rb:154:in `block (2 levels) in install_libraries'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer/xcode/pods_project_generator.rb:152:in `each'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer/xcode/pods_project_generator.rb:152:in `block in install_libraries'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/user_interface.rb:141:in `message'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer/xcode/pods_project_generator.rb:151:in `install_libraries'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer/xcode/pods_project_generator.rb:64:in `generate!'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer.rb:177:in `block in generate_pods_project'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/user_interface.rb:63:in `section'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer.rb:176:in `generate_pods_project'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/installer.rb:116:in `install!'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/command/install.rb:37:in `run'
/Users/cortman/.gem/ruby/2.2.4/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/lib/cocoapods/command.rb:50:in `run'
/Users/cortman/.gem/ruby/2.2.4/bundler/gems/CocoaPods-868991facb82/bin/pod:55:in `<top (required)>'
/Users/cortman/.gem/ruby/2.2.4/bin/pod:22:in `load'
/Users/cortman/.gem/ruby/2.2.4/bin/pod:22:in `<top (required)>'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:63:in `load'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:63:in `kernel_load'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:24:in `run'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/cli.rb:304:in `exec'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/cli.rb:11:in `start'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/exe/bundle:27:in `block in <top (required)>'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/lib/bundler/friendly_errors.rb:98:in `with_friendly_errors'
/Users/cortman/.gem/ruby/2.2.4/gems/bundler-1.12.5/exe/bundle:19:in `<top (required)>'
/Users/cortman/.gem/ruby/2.2.4/bin/bundle:23:in `load'
/Users/cortman/.gem/ruby/2.2.4/bin/bundle:23:in `<main>'
```

**Comment by chrisortman** [comment]
@mrackwitz I'm trying to figure out how to write the unit test for this but this requires a pretty thorough understanding of the way the existing unit tests work. 

Can you provide some guidance on how to set this up? 

When I compare what is generated before & after this pull request in my own project (https://github.com/chrisortman/TestGA)  it looks like the Pods-*.xcconfig is missing the 

FRAMEWORK_SEARCH_PATHS, LIBRARY_SEARCH_PATHS, OTHER_LDFLAGS settings. Presumably because without traversing the `target` as per the change there's nothing to add.

I could just set up a mock expectation that asserts the correct method is called, but that will be brittle. So I've tried looking at the _with a scoped target_ describe blocks hoping to add similar checks to what is already in the test for asserting that FRAMEWORK_SEARCH_PATHS are set, but I can't get a failing test which I think is because the test infrastructure is more correct than the implementation

**Comment by chrisortman** [comment]
I put together a test for this PR. Not sure what to do with it
See this commit https://github.com/chrisortman/CocoaPods/commit/8ea1ac16781d161dcb0887808ac1a04d2a226de4

**Comment by trevonmckay** [comment]
Any update getting this merged?

**Comment by dnkoutso** [comment]
@trevonmckay @plu this needs a rebase and some unit tests.

**Comment by chrisortman** [comment]
I think in order to make the build pass this PR will need to be accepted first? 

https://github.com/CocoaPods/cocoapods-integration-specs/pull/73

I'm not the original author of the PR, but I have a unit test written and can do the rebase if it will help move this along. I think I'd have to make a new PR then though?

**Comment by Coeur** [comment]
@dnkoutso, as a member of the project, you may have the ability to rebase during the merging process: 
screen shot 2016-11-15 at 09 50 54

**Comment by dnkoutso** [comment]
@Coeur sorry I am not a member of the project :)

**Comment by segiddins** [comment]
Nope, this needs a manual rebase

**Comment by plu** [comment]
Rebased and added @chrisortman's Unit Test.

**Comment by jafara** [comment]
Just hit that...@plu any chance this will get merged?

**Comment by plu** [comment]
Sorry, I gave up.

**Comment by jafara** [comment]
CocoaPods gods whats left for this to land? It seems there are tests for this PR

**Comment by DanToml** [comment]
@jafara gonna rebase and finish this up this week and see what's up with the failing tests.

**Comment by chrisortman** [comment]
Tests were failing because there is an open unmerged pr to change the
integration specs to reflect these changes

**Comment by DanToml** [comment]
Merged in #6287

**Comment by YANOUSHek** [comment]
I'm using 1.2.0 and this bug seems to still be present. Had to extract my test targets from the app target and append all the pods to them manually because `inherit! :search_paths` didn't add all the required search paths.

**Comment by krider2010** [comment]
I think this has regressed in 1.2.0 final - it was fixed in one of the release candidates as I tested it then and it was fine, but have just spent a frustrating morning going in circles with it!

**Comment by plu** [comment]
This broke via c45a4bb8d021de874c0e99ea0ba6d1baee61883c

**Comment by dnkoutso** [comment]
@plu can you explain? Based on https://github.com/CocoaPods/CocoaPods/issues/6065 static frameworks on static libraries should not be linked if the target is using `inherit! search_paths`.

There is one case where this is broken if the target explicitly asks for a pod to be integrated which at the moment will not properly add the `-framework` or `-l` flags.

I am working on a PR that fixes the case above.

**Comment by Coeur** [comment]
So the pull request that broke it was #6437

**Comment by plu** [comment]
Here's a sample project that shows the issue: https://github.com/plu/SearchPath

If you build the test target, it cannot find the `GoogleMobileAds` module.

Once I comment out these lines here it's working again ([L106 & L108](https://github.com/CocoaPods/CocoaPods/blob/1.2.0/lib/cocoapods/generator/xcconfig/xcconfig_helper.rb#L106-L108)):

```ruby
            #if aggregate_target.nil? || aggregate_target.target_definition.inheritance != 'search_paths'
              XCConfigHelper.add_framework_build_settings(vendored_static_framework, xcconfig, pod_target.sandbox.root)
            #end
```

After doing this change and running `pod install` again the local changes in the repository are:

```diff
diff --git a/Pods/Target Support Files/Pods-SearchPathTests/Pods-SearchPathTests.debug.xcconfig b/Pods/Target Support Files/Pods-SearchPathTests/Pods-SearchPa
index 5aa6975..4de062c 100644
--- a/Pods/Target Support Files/Pods-SearchPathTests/Pods-SearchPathTests.debug.xcconfig
+++ b/Pods/Target Support Files/Pods-SearchPathTests/Pods-SearchPathTests.debug.xcconfig
@@ -1,9 +1,9 @@
-FRAMEWORK_SEARCH_PATHS = $(inherited) "$PODS_CONFIGURATION_BUILD_DIR/1PasswordExtension"
+FRAMEWORK_SEARCH_PATHS = $(inherited) "$PODS_CONFIGURATION_BUILD_DIR/1PasswordExtension" "${PODS_ROOT}/Google-Mobile-Ads-SDK/Frameworks"
 GCC_PREPROCESSOR_DEFINITIONS = $(inherited) COCOAPODS=1
 HEADER_SEARCH_PATHS = $(inherited) "${PODS_ROOT}/Headers/Public" "${PODS_ROOT}/Headers/Public/Google-Mobile-Ads-SDK"
 LD_RUNPATH_SEARCH_PATHS = $(inherited) '@executable_path/Frameworks' '@loader_path/Frameworks'
 OTHER_CFLAGS = $(inherited) -iquote "$PODS_CONFIGURATION_BUILD_DIR/1PasswordExtension/OnePasswordExtension.framework/Headers" -isystem "${PODS_ROOT}/Headers/
-OTHER_LDFLAGS = $(inherited) -framework "AVFoundation" -framework "AudioToolbox" -framework "CoreBluetooth" -framework "CoreGraphics" -framework "CoreMedia"
+OTHER_LDFLAGS = $(inherited) -framework "AVFoundation" -framework "AudioToolbox" -framework "CoreBluetooth" -framework "CoreGraphics" -framework "CoreMedia"
 PODS_BUILD_DIR = $BUILD_DIR
 PODS_CONFIGURATION_BUILD_DIR = $PODS_BUILD_DIR/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)
 PODS_ROOT = ${SRCROOT}/Pods
diff --git a/Pods/Target Support Files/Pods-SearchPathTests/Pods-SearchPathTests.release.xcconfig b/Pods/Target Support Files/Pods-SearchPathTests/Pods-Search
index 5aa6975..4de062c 100644
--- a/Pods/Target Support Files/Pods-SearchPathTests/Pods-SearchPathTests.release.xcconfig
+++ b/Pods/Target Support Files/Pods-SearchPathTests/Pods-SearchPathTests.release.xcconfig
@@ -1,9 +1,9 @@
-FRAMEWORK_SEARCH_PATHS = $(inherited) "$PODS_CONFIGURATION_BUILD_DIR/1PasswordExtension"
+FRAMEWORK_SEARCH_PATHS = $(inherited) "$PODS_CONFIGURATION_BUILD_DIR/1PasswordExtension" "${PODS_ROOT}/Google-Mobile-Ads-SDK/Frameworks"
 GCC_PREPROCESSOR_DEFINITIONS = $(inherited) COCOAPODS=1
 HEADER_SEARCH_PATHS = $(inherited) "${PODS_ROOT}/Headers/Public" "${PODS_ROOT}/Headers/Public/Google-Mobile-Ads-SDK"
 LD_RUNPATH_SEARCH_PATHS = $(inherited) '@executable_path/Frameworks' '@loader_path/Frameworks'
 OTHER_CFLAGS = $(inherited) -iquote "$PODS_CONFIGURATION_BUILD_DIR/1PasswordExtension/OnePasswordExtension.framework/Headers" -isystem "${PODS_ROOT}/Headers/
-OTHER_LDFLAGS = $(inherited) -framework "AVFoundation" -framework "AudioToolbox" -framework "CoreBluetooth" -framework "CoreGraphics" -framework "CoreMedia"
+OTHER_LDFLAGS = $(inherited) -framework "AVFoundation" -framework "AudioToolbox" -framework "CoreBluetooth" -framework "CoreGraphics" -framework "CoreMedia"
 PODS_BUILD_DIR = $BUILD_DIR
 PODS_CONFIGURATION_BUILD_DIR = $PODS_BUILD_DIR/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)
 PODS_ROOT = ${SRCROOT}/Pods
```

**Comment by plu** [comment]
I believe that `def self.add_framework_build_settings` must be split into two methods. One enhancing the `FRAMEWORK_SEARCH_PATHS`, and another one enhancing `OTHER_LDFLAGS`. For the `search_paths` use case, always call the method enhancing `FRAMEWORK_SEARCH_PATHS`.

**Comment by dnkoutso** [comment]
Yes you are correct. I'll be opening a PR that fixes this.

**Comment by plu** [comment]
Thank you!

**Comment by plu** [comment]
I can also test on a more complex project once you have a patch ready.

**Comment by dnkoutso** [comment]
@plu https://github.com/CocoaPods/CocoaPods/pull/6477

**Comment by dnkoutso** [comment]
@plu I tried it on the project and it fixes it. Would be good to get a more complicated project :)

**Comment by YANOUSHek** [comment]
I can test this on a more complicated project (3 framework targets, each with unit tests and an app target with unit and UI tests; all in one project), but I'd need some instructions on how to get the dev version working on my machine.

**Comment by dnkoutso** [comment]
@YANOUSHek if you have a `Gemfile` then just add:

```
gem 'cocoapods', github: 'dnkoutso/CocoaPods', branch: 'better-static-linking'
```

Or you can choose the master branch now since it was merged.

**Comment by YANOUSHek** [comment]
@dnkoutso I think I've managed to get the current master branch installed and unfortunately it does not solve the problem. When I migrated my Podfile to use search path inheritance it still doesn't add the static libs into the project.

I'm not really sure if I'm running the version from master (they both report the same version number – 1.2.0) but I'm pretty sure I managed to get the newest one.

**Comment by dnkoutso** [comment]
> " it still doesn't add the static libs into the project."

what do you mean by that? Its not supposed to add pre-built static libraries or frameworks to your project.

Can you please open a new issue with a sample project instead so we don't post on a closed issue.

**Comment by krider2010** [comment]
A simple test case that now fails (again, after working in 1.2 beta 3) is to include Google Analytics into things. It fails to build any test target with the inherit paths option set. This regression is seriously problematic. Especially after it took so long to be solved in the first place. I had the luxery of it working for maybe a month at most 😞  I know this issue is closed, but a test case or something is needed to support it not happening again.

**Comment by YANOUSHek** [comment]
@dnkoutso I've created #6479 with all the examples I could have taught of.

**Comment by dnkoutso** [comment]
@krider2010 tests were added and this was further improved and merged to master. It will most likely ship with 1.2.1.

**Comment by miktav** [comment]
@krider2010 I had the same problem with Google Analytics, where my test target failed to build. I posted a question about this on [stackoverflow](http://stackoverflow.com/questions/42934352/ld-framework-not-found-gglanalytics-for-architecture-x86-64-error-on-test-tar) before I saw this thread. The good news is that after upgrading to Cocoapods 1.2.1.beta.1, and re-running pod install, my test target now builds.

**Comment by dnkoutso** [comment]
Yeap this is fixed in 1.2.1.beta.1 thanks for verifying again.

**Comment by miktav** [comment]
@dnkoutso Thank you so much for resolving this.

**Comment by Oyashirox** [comment]
Hey, thank you for this, I can now run my build ! I still have a warning though : 
`ld: warning: Auto-Linking supplied '/path/Pods/GoogleMaps/Maps/Frameworks/GoogleMaps.framework/GoogleMaps', framework linker option at /path/Pods/GoogleMaps/Maps/Frameworks/GoogleMaps.framework/GoogleMaps is not a dylib`

**Comment by dnkoutso** [comment]
@Oyashirox please do not respond to a close issue. File a new one and provide a sample project.

**Pull request closed without merging**